tests/unit/test_svs_tiered.cpp

```diff
@@ -0,0 +1,2416 @@
+    // Brute-force index checks for timeout after each vector.
+    size_t checks_in_flat = flat_index->indexSize();
+    VecSimQueryParams qparams = {.timeoutCtx = &checks_in_flat};
+    VecSim_SetTimeoutCallbackFunction([](void *ctx) {
+        auto count = static_cast<size_t *>(ctx);
+        if (*count == 0) {
+            return 1;
+        }
+        (*count)--;
+        return 0;
+    });
+    res = VecSimIndex_TopKQuery(tiered_index, query_0, k, &qparams, BY_SCORE);
+    ASSERT_TRUE(res->results.empty());
+    ASSERT_EQ(VecSimQueryReply_GetCode(res), VecSim_QueryReply_TimedOut);
+    VecSimQueryReply_Free(res);
+    // Make sure we didn't get the timeout in the flat index.
+    checks_in_flat = flat_index->indexSize(); // Reset the counter.
+    res = VecSimIndex_TopKQuery(flat_index, query_0, k, &qparams, BY_SCORE);
+    ASSERT_EQ(VecSimQueryReply_GetCode(res), VecSim_QueryReply_OK);
+    VecSimQueryReply_Free(res);
+
+    // Clean up.
+    VecSim_SetTimeoutCallbackFunction([](void *ctx) { return 0; });
+}
+
+TYPED_TEST(SVSTieredIndexTest, deleteVector) {
+    // Create TieredSVS index instance with a mock queue.
+    size_t dim = 4;
+    SVSParams params = {.type = TypeParam::get_index_type(),
+                        .dim = dim,
+                        .metric = VecSimMetric_L2,
+                        .num_threads = 1};
+    VecSimParams svs_params = CreateParams(params);
+    auto mock_thread_pool = tieredIndexMock();
+
+    auto *tiered_index = this->CreateTieredSVSIndex(svs_params, mock_thread_pool, 1);
+    ASSERT_INDEX(tiered_index);
+    auto allocator = tiered_index->getAllocator();
+
+    labelType vec_label = 0;
+    // Delete from an empty index.
+    ASSERT_EQ(tiered_index->deleteVector(vec_label), 0);
+
+    // Create a vector and add it to the tiered index (expect it to go into the flat buffer).
+    TEST_DATA_T vector[dim];
+    GenerateVector<TEST_DATA_T>(vector, dim, vec_label);
+    VecSimIndex_AddVector(tiered_index, vector, vec_label);
+    ASSERT_EQ(tiered_index->indexSize(), 1);
+    ASSERT_EQ(tiered_index->GetFlatIndex()->indexSize(), 1);
+
+    // Remove vector from flat buffer.
+    ASSERT_EQ(tiered_index->deleteVector(vec_label), 1);
+    ASSERT_EQ(tiered_index->indexSize(), 0);
+    ASSERT_EQ(tiered_index->GetFlatIndex()->indexSize(), 0);
+
+    mock_thread_pool.thread_iteration();
+    ASSERT_EQ(tiered_index->GetBackendIndex()->indexSize(), 0);
+
+    // Create a vector and add it to SVS in the tiered index.
+    VecSimIndex_AddVector(tiered_index->GetBackendIndex(), vector, vec_label);
+    ASSERT_EQ(tiered_index->indexSize(), 1);
+    ASSERT_EQ(tiered_index->GetBackendIndex()->indexSize(), 1);
+
+    // Remove from main index.
+    ASSERT_EQ(tiered_index->deleteVector(vec_label), 1);
+    ASSERT_EQ(tiered_index->indexLabelCount(), 0);
+    ASSERT_EQ(tiered_index->indexSize(), 0);
+    // ASSERT_EQ(tiered_index->getSVSIndex()->getNumMarkedDeleted(), 1);
```

Comment:
why is it commented out?

Reply:
Copy-paste artefact from HNSW test
Removed.